src/main/java/org/threadly/concurrent/PrioritySchedulerStatisticTracker.java

```diff
@@ -769,50 +606,207 @@ private static double getTruePercent(Collection<Boolean> list) {
+      highPriorityWorkerAvailable = new ConcurrentArrayList<Boolean>(0, endPadding);
+      lowPriorityExecutionDelay = new ConcurrentArrayList<Long>(0, endPadding);
+      highPriorityExecutionDelay = new ConcurrentArrayList<Long>(0, endPadding);
+    }
+
+    /**
+     * Called at the start of execution to track statistics around task execution.
+     * 
+     * @param taskWrapper Wrapper that is about to be executed
+     */
+    protected void trackTaskStart(Wrapper taskWrapper) {
+      runningTasks.put(taskWrapper, taskWrapper.startTime = Clock.accurateForwardProgressingMillis());
+      
+      switch (taskWrapper.priority) {
+        case High:
+          totalHighPriorityExecutions.incrementAndGet();
+          break;
+        case Low:
+          totalLowPriorityExecutions.incrementAndGet();
+          break;
+        default:
+          throw new UnsupportedOperationException();
+      }
+    }
+    
+    /**
+     * Used to track how long tasks are tacking to complete.
+     * 
+     * @param taskWrapper wrapper for task that completed
+     */
+    protected void trackTaskFinish(Wrapper taskWrapper) {
+      long finishTime = Clock.accurateForwardProgressingMillis();
+      synchronized (runTimes.getModificationLock()) {
+        runTimes.add(finishTime - taskWrapper.startTime);
+        trimList(runTimes);
+      }
+      runningTasks.remove(taskWrapper);
+    }
     
-    switch (taskWrapper.priority) {
-      case High:
-        totalHighPriorityExecutions.incrementAndGet();
-        break;
-      case Low:
-        totalLowPriorityExecutions.incrementAndGet();
-        break;
-      default:
-        throw new UnsupportedOperationException();
+    /**
+     * Reduces the list size to be within the max window size.
+     * 
+     * Should have the list synchronized/locked before calling.
+     * 
+     * @param list LinkedList to check size of.
+     */
+    @SuppressWarnings("rawtypes")
+    protected static void trimList(Deque list) {
+      while (list.size() > MAX_WINDOW_SIZE) {
+        list.removeFirst();
+      }
     }
   }
   
   /**
-   * Used to track how long tasks are tacking to complete.
+   * <p>An extending class of {@link WorkerPool}, allowing us to gather statistics about how 
+   * workers are used in the executor.  An example of such statistics are how long tasks are 
+   * delayed from their desired execution.  Another example is how often a worker can be reused vs 
+   * how often they have to be created.</p>
    * 
-   * @param taskWrapper wrapper for task that completed
+   * @author jent - Mike Jensen
+   * @since 3.5.0
    */
-  protected void trackTaskFinish(Wrapper taskWrapper) {
-    long finishTime = Clock.accurateForwardProgressingMillis();
-    synchronized (runTimes.getModificationLock()) {
-      runTimes.add(finishTime - taskWrapper.startTime);
-      trimList(runTimes);
+  protected static class StatisticWorkerPool extends WorkerPool {
```

Comment:
We must extend the normal WorkerPool to be able to collect thread usage statistics from run(High/Low)PriorityTask.  Previously we were just overriding the functions in the parent PriorityScheduler, but that is no longer possible.  This is also why the protected constructor was made to provide the WorkerPool (so it can be extended).

You will see a similar pattern in the Profiler.